src/main/kotlin/interpreter/pyobject/types/PyBytesType.kt

```diff
@@ -0,0 +1,46 @@
+package green.sailor.kython.interpreter.pyobject.types
+
+import green.sailor.kython.interpreter.Exceptions
+import green.sailor.kython.interpreter.functions.PyBuiltinFunction
+import green.sailor.kython.interpreter.iface.ArgType
+import green.sailor.kython.interpreter.iface.PyCallableSignature
+import green.sailor.kython.interpreter.pyobject.*
+import green.sailor.kython.interpreter.throwKy
+import java.lang.IndexOutOfBoundsException
+
+object PyBytesType : PyType("bytes") {
+    override fun newInstance(kwargs: Map<String, PyObject>): PyObject {
+        val value = kwargs["value"] ?: error("Built-in signature mismatch")
+        when (value) {
+            is PyString -> {
+                return PyBytes(value.wrappedString.toByteArray())
+            }
+            else -> error("Not yet supported")
+        }
+    }
+
+    override val signature: PyCallableSignature by lazy {
+        PyCallableSignature(
+            "value" to ArgType.POSITIONAL
+        )
+    }
+
+    /** bytes.__getitem__ */
+    val pyBytesIndex = PyBuiltinFunction.wrap("__getitem__", PyCallableSignature("index" to ArgType.POSITIONAL)) {
+        val self = it["self"]!!.cast<PyBytes>()
+        val index = it["index"]!!.cast<PyInt>()
+        // TODO: PySlice check
+        try {
+            PyInt(self.wrapped[index.wrappedInt.toInt()].toLong())
+        } catch (e: IndexOutOfBoundsException) {
+            Exceptions.INDEX_ERROR("index out of range").throwKy()
```

Comment:
Change this to `Index ${index.wrappedInt} out of range`

Reply:
Shouldn't we aim to keep these consistent with cpython's errors?

Reply:
I've already decided that no, we shouldn't, because cpython errors are often unhelpful and if you do direct checks like these your code will break if cpy ever gets around to better errors.